src/PermissionMatcher.php

```diff
@@ -12,33 +12,42 @@ class PermissionMatcher
      *
      * @param array|object $permissions
      * @param array        $authzGroups
+     * @param boolean      $reverse     Returns an array where the priviliges are the keys and authzgroups the values
      * @return array
      */
-    public function match($permissions, array $authzGroups)
+    public function match($permissions, array $authzGroups, $reverse = false)
     {
         $privileges = [];
 
         foreach ($permissions as $permissionAuthzGroup => $permissionPrivileges) {
-            if ($this->hasMatchingAuthzGroup($permissionAuthzGroup, $authzGroups)) {
+            $matchingAuthzGroup = $this->hasMatchingAuthzGroup($permissionAuthzGroup, $authzGroups);
+            
+            if (!$matchingAuthzGroup) {
+                continue;
+            }
+            
+            if ($reverse) {
+                $privileges = $this->addAuthzGroupsToPrivileges($privileges, $permissionPrivileges, [$permissionAuthzGroup, $matchingAuthzGroup]);
+            } else {
                 $privileges[] = $permissionPrivileges;
             }
         }
 
-        return $this->flatten($privileges);
+        return $reverse ? $privileges : $this->flatten($privileges);
     }
 
-
     /**
      * Check if one of the authz groups match
      *
      * @param string $permissionAuthzGroup
      * @param array  $authzGroups
+     * @return string|boolean
```

Comment:
Method starting with `has` or `is` are expected to return a boolean. You should rename the function to `getMatchingAuthzGroup()`. Also consider returning `null` rather than `false` if there is no match.